### PR TITLE
Fix spurious Durable Executor test failure on killed owner

### LINKED ISSUE #10033: DurableRetrieveResultTest.testSingleExecution_WhenMigratedAfterCompletion_WhenOwnerMemberKilled
**master**
```
java.lang.AssertionError: expected:<1> but was:<2>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.durableexecutor.DurableRetrieveResultTest.testSingleExecution_WhenMigratedAfterCompletion_WhenOwnerMemberKilled(DurableRetrieveResultTest.java:142)
```
https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/7442/testReport/junit/com.hazelcast.durableexecutor/DurableRetrieveResultTest/testSingleExecution_WhenMigratedAfterCompletion_WhenOwnerMemberKilled/

### PR DESCRIPTION
**Pull request by tkountis**
Suspected that this could be due to a race between PutResult and Shutdown.
A delay filter was added on the `MockConnectionManager` and verified that it fails almost always even in my local environment.

Sleep was added to optimistically wait until the result is replicated before killing the node.

Fix #10033

### CODE REVIEW

**Comment by Donnerbart** [comment]
run-lab-run

File: hazelcast/src/test/java/com/hazelcast/durableexecutor/DurableRetrieveResultTest.java [hunk 1]
```diff
@@ -117,30 +116,41 @@ public void testRetrieveAndDispose_WhenOwnerMemberDown() throws Exception {
...
 
     @Test
-    @Ignore(value = "https://github.com/hazelcast/hazelcast/issues/10033")
     public void testSingleExecution_WhenMigratedAfterCompletion_WhenOwnerMemberKilled() throws Exception {
-        String name = randomString();
         TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(3);
-        HazelcastInstance instance1 = factory.newHazelcastInstance();
-        HazelcastInstance instance2 = factory.newHazelcastInstance();
-        factory.newHazelcastInstance();
+        factory.newInstances();
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/durableexecutor/DurableRetrieveResultTest.java]: You can shorten this to:
```java
        HazelcastInstance[] instances = factory.newInstances();
        HazelcastInstance first = instances[0];
        HazelcastInstance second = instances[1];

        waitAllForSafeState(instances);
```

**Comment by tkountis** [comment]
run-lab-run
